Notebooks/SentinelUtilities/SentinelPortal/bookmark_helper.py

```diff
@@ -0,0 +1,318 @@
+
+    @property
+    def notes(self):
+        """ notes getter """
+        return self._notes
+
+    @notes.setter
+    def notes(self, notes):
+        """ notes setter """
+        self._notes = notes
+
+    @property
+    def eventTime(self):
+        """ eventTime """
+        return self._eventTime
+
+    @eventTime.setter
+    def eventTime(self, event_time):
+        """ eventTime setter """
+        self._eventTime = event_time
+
+    @property
+    def queryStartTime(self):
+        """ queryStartTime """
+        return self._queryStartTime
+
+    @queryStartTime.setter
+    def queryStartTime(self, query_start_time):
+        """ queryStartTime setter """
+        self._queryStartTime = query_start_time
+
+    @property
+    def queryEndTime(self):
+        """ queryEndTime """
+        return self._queryEndTime
+
+    @queryEndTime.setter
+    def queryEndTime(self, query_end_time):
+        """ queryEndTime setter """
+        self._queryEndTime = query_end_time
+
+class BookmarkModel():
+    """ This class holds data model for Bookmark """
+
+    def __init__(self,
+                 bookmark_name,
+                 subscription_id,
+                 resource_group_name,
+                 workspace_name,
+                 bookmark_properties):
+        self.name = bookmark_name
+        self.type = Constants.TYPE
+        self.etag = Constants.ETAG
+        self.properties = bookmark_properties
+        self.bookmark_resource_base = Constants.BOOKMARK_RESOURCE_BASE.format(subscription_id, resource_group_name, workspace_name)
+        self.id = self.bookmark_resource_base + Constants.BOOKMARK_ID.format(bookmark_properties.bookmarkId)
+
+    @property
+    def id(self):
+        """ id getter """
+        return self._id
+
+    @id.setter
+    def id(self, bookmark_resource_id):
+        """ id setter """
+        self._id = bookmark_resource_id
+
+    @property
+    def name(self):
+        """ name getter """
+        return self._name
+
+    @name.setter
+    def name(self, bookmark_name):
+        """ name setter """
+        InputValidation.validate_input('bookmark_name', bookmark_name)
+        self._name = bookmark_name
+
+    @property
+    def type(self):
+        """ type getter """
+        return self._type
+
+    @type.setter
+    def type(self, t):
+        """ type setter """
+        self._type = t
+
+    @property
+    def etag(self):
+        """ etag getter """
+        return self._etag
+
+    @etag.setter
+    def etag(self, etag):
+        """ etag setter """
+        self._etag = etag
+
+    @property
+    def properties(self):
+        """ properties getter """
+        return self._properties
+
+    @properties.setter
+    def properties(self, bookmark_properties):
+        """ properties setter """
+        InputValidation.validate_input('bookmark_properties', bookmark_properties)
+        self._properties = bookmark_properties
+
+# pylint: disable-msg=W0703
+class BookmarkHelper:
```

Comment:
Wouldn't it make more sense to have BookmarkHelper take an access token in __init__ and store that as a attrib? Then supply a Bookmark to add_bookmark, etc. as a method parameter?

Reply:
Agreed, good point.   Changed.